src/www/firewall_rules_edit.php

```diff
@@ -349,13 +349,20 @@ function is_posnumericint($arg) {
         $input_errors[] = gettext("If you specify TCP flags that should be set you should specify out of which flags as well.");
 
 
+    // must use value + 1 in comparison due to not being able to store "0" in rule config
+    if (!empty($pconfig['priority']) && ($pconfig['priority'] < 1 || $pconfig['priority'] > 8))
+        $input_errors[] = gettext('Priority must be between 0 and 7.');
```

Comment:
From the error texts, make clear which value is wrong. This isn't unlikely to hit since the value is provided via select box, but still...